Mathlib/Geometry/Group/Marking.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Analysis.Normed.Group.Basic
+import Mathlib.GroupTheory.FreeGroup.Reduce
+
+/-!
+# Marked groups
+
+This file defines group markings and induces a norm on marked groups.
+
+## Main declarations
+
+* `GroupMarking G S`: Marking of the group `G` by a type `S`, namely a surjective monoid
+  homomorphism `FreeGroup S →* G`.
+* `MarkedGroup`: If `m : GroupMarking G S`, then `MarkedGroup m` is a type synonym for `G`
+  endowed with the metric coming from `m`.
+* `MarkedGroup.instNormedGroup`: A marked group is normed by its marking.
```

Comment:
It would be nice to give here a clearer description of what the metric and norm actually are. Currently the documentation only says that a norm is induced, with no clue as to how they are defined, or what they mean.